src/main/java/com/couchbase/intellij/database/ActiveCluster.java

```diff
@@ -94,14 +105,37 @@ public void connect(SavedCluster savedCluster, Runnable disconnectListener) {
 
                     eventBus.stop(Duration.ZERO);
                     disconnect();
+                    ActiveCluster.getInstance().get().environment().shutdown();
                 }
             });
+
             this.cluster = cluster;
             this.savedCluster = savedCluster;
             this.password = password;
             if (savedCluster.getColor() != null) {
                 this.color = Color.decode(savedCluster.getColor());
             }
+
+            ServerOverview overview = CouchbaseRestAPI.getOverview();
+            setServices(overview.getNodes().stream()
+                    .flatMap(node -> node.getServices().stream()).distinct().collect(Collectors.toList()));
+
+            setVersion(overview.getNodes().get(0).getVersion()
+                    .substring(0, overview.getNodes().get(0).getVersion().indexOf('-')));
+
```

Comment:
It is mandatory to get the available services before saying that we connected to the cluster